<?xml version="1.0"?>
<robot xmlns:xacro="http://wiki.ros.org/xacro" name="$(arg ar_model)">
  <xacro:arg name="serial_port" default="/dev/ttyACM0"/>
  <xacro:arg name="arduino_serial_port" default="/dev/ttyUSB0"/>
  <xacro:arg name="calibrate" default="True"/>
  <xacro:arg name="include_gripper" default="true"/>
  <xacro:include filename="$(find ar_description)/urdf/ar_macro.xacro"/>
  <xacro:include filename="$(find ar_description)/urdf/ar.ros2_control.xacro"/>
  
  <link name="world" />
  <xacro:ar_robot
    parent="world"
    robot_parameters_file="$(find ar_description)/config/$(arg ar_model).yaml"
  >
    <origin xyz="0 0 0" rpy="0 0 0" />
  </xacro:ar_robot>

  <xacro:ar_ros2_control
    ar_model="$(arg ar_model)"
    plugin_name="ar_hardware_interface/ARHardwareInterface"
    serial_port="$(arg serial_port)"
    calibrate="$(arg calibrate)"
    robot_parameters_file="$(find ar_description)/config/$(arg ar_model).yaml"
    joint_offset_parameters_file="$(find ar_hardware_interface)/config/joint_offsets/$(arg ar_model).yaml"
    driver_parameters_file="$(find ar_hardware_interface)/config/driver.yaml"
  />

  <xacro:if value="$(arg include_gripper)">
    <xacro:include filename="$(find ar_description)/urdf/ar_gripper_macro.xacro"/>
    <xacro:include filename="$(find ar_description)/urdf/ar_gripper.ros2_control.xacro"/>

    <xacro:ar_gripper parent="ee_link" />

    <xacro:ar_gripper_ros2_control
      name="ar_servo_gripper"
      plugin_name="ar_hardware_interface/ARServoGripperHWInterface"
      serial_port="$(arg arduino_serial_port)"
    />
  </xacro:if>

</robot>
